### ISSUE TITLE
discussion: return a class instead of the React.createElement result

### ISSUE THREAD

**Issue by phated** [omniscientjs/omniscient#46]
I've been diving into (and loving) this project for the last week.  I was just wondering what your thoughts were towards React 0.13 and dealing with generic classes.  Would is be possible to return the custom class from the internals instead of returning the result of `React.createElement`?  I ask because it would cleanup the jsx usage.  The other option I was thinking about was making it a config option now that `withDefaults` is an available API.

Anyhow, keep up the excellent work.

**Comment by mikaelbr** [comment]
Hi and thanks for great feedback!

I did some initial testing, switching over to use module pattern, when React 0.13 came out. But it didn't pan out straight away, and I postponed further testing for fixing some other issues. If we switched to not using `React.createElement` we would have to switch over to doing something manually, though. Such as handling mixins and order of lifecycle methods (asserting that ever lifecycle method is called), and we would need to handle better shouldComponentUpdate (as it is used as a mixin now). 

It should definitely be doable, but it might require doing more of the work in Omniscient instead of having React do it for us - and if React plans on still supporting `React.createElement` it might as well be that using `React.createElement` would be the best.

But it's defiantly worth checking out and experimenting with before we do any major release of Omniscient (planned within a week or so). `.jsx` is not a perfect solution so we should try to get rid of it. 

I never thought about having `.jsx` as a part of `.withDefaults()`. That's an excellent idea, and something we could do right now for easier use with projects that only uses the jsx syntax!
